buffer/src/main/java/io/netty/buffer/api/BufferAllocator.java

```diff
@@ -133,6 +135,57 @@ static BufferAllocator offHeapPooled() {
      */
     Supplier<Buffer> constBufferSupplier(byte[] bytes);
 
+    /**
+     * Allocate a {@link Buffer} with the same size and contents of the given byte array.
+     * The allocated buffer is <em>NOT</em> backed by the given byte array, and changes to the contents of either
+     * will not be reflected in the other.
+     * This may throw an {@link OutOfMemoryError} if there is not enough free memory available to allocate a
+     * {@link Buffer} of the requested size.
+     * <p>
+     * The allocated buffer will use big endian byte order.
+     *
+     * @param bytes The byte array that determines the size and contents of the new buffer.
+     * @return The newly allocated {@link Buffer}.
+     * @throws IllegalStateException if this allocator has been {@linkplain #close() closed}.
+     */
+    default Buffer copyOf(byte[] bytes) {
+        return allocate(bytes.length).writeBytes(bytes);
+    }
+
+    /**
+     * Allocate a {@link Buffer} with the same size and contents, as the contents of the given {@link ByteBuffer}.
+     * The allocated buffer is <em>NOT</em> backed by the give byte buffer, and changes to the contents of either
+     * will not be reflected in the other.
+     * The position and limit of the given byte buffer defines the region of contents that will be copied.
+     * The position and limit are not modified by this method.
+     * This may throw an {@link OutOfMemoryError} if there is not enough free memory available to allocate a
+     * {@link Buffer} of the requested size.
+     * <p>
+     * The allocated buffer will use big endian byte order, regardless of the byte order of the given buffer.
+     * The contents of the byte buffer will be copied without regard to the byte order, as if the bytes are copied
+     * one at a time.
+     *
+     * @param buffer The byte buffer, whose contents determine the capacity and contents of the allocated buffer.
+     * @return The newly allocated {@link Buffer}.
+     * @throws IllegalStateException if this allocator has been {@linkplain #close() closed}.
+     */
+    default Buffer copyOf(ByteBuffer buffer) {
+        if (!buffer.hasRemaining()) {
+            return allocate(0);
+        }
+        final Buffer copy = allocate(buffer.remaining());
+        final ByteBuffer duplicate = buffer.duplicate();
+        copy.forEachWritable(0, (i, component) -> {
+            ByteBuffer dest = component.writableBuffer();
+            int length = Math.min(dest.capacity(), duplicate.remaining());
+            Statics.bbput(dest, 0, duplicate, duplicate.position(), length);
+            duplicate.position(length + duplicate.position());
+            return true;
+        });
+        copy.skipWritable(copy.capacity());
```

Comment:
is it guaranteed that `copy.capacity()` == `buffer.remaining()` ? I would assume that depending the allocator the capacity()  could also be larger. So maybe it is safer and more correct to first store `buffer.remaining()` and use the value for `skipWritable()`